### PR TITLE
include multi-stack version setup in registry guide

### PR DESCRIPTION
**Pull request by yangcao77**
Signed-off-by: Stephanie <yangcao@redhat.com>

For issue https://github.com/devfile/api/issues/810

update the registry guide to include setup for multiple stack versions in a registry. 
(The design doc is https://docs.google.com/document/d/1_eXbAV1eKetKKu_FHGRw6kUUiUhcp7j1gB0uUrG0WF8/edit?usp=sharing)

added pages for adding a stack.yaml file, for creating a stack. updated existing index schema page.

### CODE REVIEW

File: docs/modules/user-guide/partials/proc_adding-a-registry-schema.adoc [hunk 1]
```diff
@@ -107,45 +165,80 @@ The `index.json` file contains metadata for the stacks and samples in the regist
...
 
 | `description`
 | `string`
-| The description of a devfile.
+| The description of a stack.
 
 | `type`
 | `DevfileType`
-| The type of a devfile that currently supports stacks and samples.
+| The type: stack/sample.
```
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_adding-a-registry-schema.adoc:186]: To be consisted with all the other phrasing, I would remove the colon. 

Does it make just as much sense to say, `The type of a stack or sample.`

File: docs/modules/user-guide/partials/proc_creating-a-devfile-stack.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[id="creating-a-devfile-stack_{context}"]
+= Creating a Devfile stack
+
+[role="_abstract"]
+Create a devfile stack to package into the devfile registry.
```
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_creating-a-devfile-stack.adoc:5]: Can we include a value or purpose behind creating a devfile stack to package into the registry? 

Like, `Create a devfile stack to package into the devfile registry so you can <state value.>` 

Would this work?:

`Create a devfile stack to package into the devfile registry so you can quickly access, share, and duplicate your different stacks for your various projects.`

File: docs/modules/user-guide/partials/proc_creating-a-devfile-stack.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+.Procedure
+
+. Create a stack folder with a name matches the stack name, for example, `java-wildfly/`.
```
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_creating-a-devfile-stack.adoc:9]: `Create a stack folder with a name that matches the stack name. For example, `java-wildfly/`.`

Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_creating-a-devfile-stack.adoc:9]: I think it reads better with the `that` included and a separate sentence for the example :)

**Review by jc-berger — APPROVED** [review]
/lgtm

**Comment by yangcao77** [comment]
The errors from CI are for docs whose content were not updated. The error from the new added doc is suggesting to change `yaml` to  `YAML`, which does not make sense in the content. e.g. `stack.yaml`
Merging as a result.

**Pull request merged**